client/src/components/Topics/Topics.tsx

```diff
@@ -0,0 +1,44 @@
+import { LineLoader } from 'components/Loader';
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { getTopics, Topic } from 'utils/apiRequests';
+import Notification from 'utils/Notification';
+import { titleToUrl } from 'utils/url';
+import TopicCard from './TopicCard';
+
+interface Props {}
+
+const Topics: React.FC<Props> = () => {
+
+  const [topics, setTopics] = useState<Topic[]>([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    getTopics().then(setTopics).catch(Notification.error);
+  }, []);
+
+  const handleClick = (topic: Topic) => {
+    navigate(`/visualiser/${titleToUrl(topic.title)}`);
+  };
+
+  return topics && topics.length > 0 ? (
+    <div
+      style={{display: 'flex', flexWrap: 'wrap', flexDirection: 'row', justifyContent: 'center', marginTop: '5%', gap: '20px 50px'}}>
+      {topics.map((topic, idx) => (
+        <div
```

Comment:
There will be a bit of a div soup happening here. Can you define the onClick handler inside the TopicCard? Also, is it possible to just change the topic card to an \<a\> component (add "display: block" styling) whose href is /visualiser/....... etc.?